yash-datalake/azuredeploy.json

```diff
@@ -0,0 +1,576 @@
+        "No"
+      ],
+      "defaultValue": "Yes",
+      "metadata": {
+        "description": "Select whether the VM should be in production or not."
+      }
+    },
+    "dataFactoryName": {
+      "type": "string",
+      "metadata": {
+        "description": "Name of the data factory. Must be globally unique."
+      },
+      "defaultValue": "[toLower(uniqueString(resourceGroup().id, 'datafactorya'))]"
+    },
+    "servicePrincipalId": {
+      "type": "string",
+      "defaultValue": "null",
+      "metadata": {
+        "description": "The ID of the service principal that has permissions to create HDInsight clusters in your subscription."
+      }
+    },
+    "servicePrincipalKey": {
+      "type": "string",
+      "defaultValue": "null",
+      "metadata": {
+        "description": "The access key of the service principal that has permissions to create HDInsight clusters in your subscription."
+      }
+    },
+    "dataLakeAnalyticsLocation": {
+      "type": "string",
+      "allowedValues": [
+        "East US 2",
+        "North Europe",
+        "Central US",
+        "West Europe",
+        "Australia East"
+      ],
+      "defaultValue": "East US 2",
+      "metadata": {
+        "description": "The location in which the resources will be created.Check supported locations"
+      }
+    },
+    "_artifactsLocation": {
+      "type": "string",
+      "defaultValue": "https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/yash-datalake/",
+      "metadata": {
+        "description": "The base URI where artifacts required by this template are located here"
+      }
+    },
+    "_artifactsLocationSasToken": {
+      "type": "secureString",
+      "defaultValue": "",
+      "metadata": {
+        "description": "The sasToken required to access _artifactsLocation.  When the template is deployed using the accompanying scripts, a sasToken will be automatically generated. Use the defaultValue if the staging location is not secured."
+      }
+    }
+  },
+  "variables": {
+    "storageAccountName": "[toLower(uniqueString(resourceGroup().id, 'storagea'))]",
+    "blobContainer": "raw",
+    "storageAccountType": "Standard_LRS",
+    "maximumThroughputUnits": 8,
+    "DataCleaningInputStream": "datalakerawstreama",
+    "DataCleaningOutputStream": "datalakecleanstreama",
+    "captureNameFormat": "{Namespace}/{EventHub}/{PartitionId}/{Year}/{Month}/{Day}/{Hour}/{Minute}/{Second}",
+    "DataCleaningStreamAnalyticsJob": "cleanOrdersStreamAnalyticsJoba",
+    "DataAggregatorStreamAnalyticsJob1": "sellingProductPerMinJoba",
+    "DataAggregatorStreamAnalyticsJob2": "geoLocationsJoba",
+    "DataAggregatorStreamAnalyticsJob3": "ageGroupWiseRevenueJoba",
+    "appName": "[toLower(uniqueString(resourceGroup().id, 'azuredatalakeuiappa'))]",
+    "dnsLabelPrefix": "[concat('datalake',uniquestring(resourceGroup().id,deployment().name))]",
+    "ES_Function_name": "elkstackfuntiona",
+    "serverfarms_name": "EastUS2Plana",

```

Comment:
this needs to be unique within the subscription

Reply:
Updated file with required changes